extensions/vault/deployment/src/main/java/io/quarkus/vault/deployment/DevServicesVaultProcessor.java

```diff
@@ -0,0 +1,202 @@
+package io.quarkus.vault.deployment;
+
+import java.io.Closeable;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.OptionalInt;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.jboss.logging.Logger;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.vault.VaultContainer;
+
+import io.quarkus.bootstrap.classloading.QuarkusClassLoader;
+import io.quarkus.deployment.IsDockerWorking;
+import io.quarkus.deployment.IsNormal;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.LaunchModeBuildItem;
+import io.quarkus.deployment.builditem.RunTimeConfigurationDefaultBuildItem;
+import io.quarkus.deployment.builditem.ServiceStartBuildItem;
+import io.quarkus.runtime.LaunchMode;
+import io.quarkus.runtime.configuration.ConfigUtils;
+import io.quarkus.vault.deployment.devservices.DevServicesVaultResultBuildItem;
+import io.quarkus.vault.runtime.config.DevServicesConfig;
+import io.quarkus.vault.runtime.config.VaultBuildTimeConfig;
+
+public class DevServicesVaultProcessor {
+    private static final Logger log = Logger.getLogger(DevServicesVaultProcessor.class);
+    private static final String VAULT_IMAGE = "vault:1.7.1";
+    private static final int VAULT_EXPOSED_PORT = 8200;
+    private static final String CONFIG_PREFIX = "quarkus.vault.";
+    private static final String URL_CONFIG_KEY = CONFIG_PREFIX + "url";
+    private static final String AUTH_CONFIG_PREFIX = CONFIG_PREFIX + "authentication.";
+    private static final String CLIENT_TOKEN_CONFIG_KEY = AUTH_CONFIG_PREFIX + "client-token";
+    private static volatile List<Closeable> closeables;
+    private static volatile DevServicesConfig capturedDevServicesConfiguration;
+    private static volatile boolean first = true;
+    private final IsDockerWorking isDockerWorking = new IsDockerWorking(true);
+
+    @BuildStep(onlyIfNot = IsNormal.class)
+    public DevServicesVaultResultBuildItem startVaultContainers(LaunchModeBuildItem launchMode,
+            BuildProducer<RunTimeConfigurationDefaultBuildItem> runTimeConfiguration,
+            BuildProducer<ServiceStartBuildItem> serviceStartBuildItemBuildProducer, VaultBuildTimeConfig config) {
+
+        DevServicesConfig currentDevServicesConfiguration = config.devservices;
```

Comment:
may be we should be consistent on naming: `devServices` instead of `devservices`

Reply:
The configuration key `devservices` (all lowercase) is consistent with all the other DevServices configuration keys (e.g. Datasources, Mongo, Redis, etc.).